### PR TITLE
Add interpose

### PR DESCRIPTION
**Pull request by tgriesser**

### CODE REVIEW

**Comment by kevinbeaty** [comment]
I like this.  Would be useful for splitting and rejoining lines in a stream as well.  Can you add it to the Readme?

File: src/transducers.js [hunk 1]
```diff
@@ -214,3 +214,25 @@ class Tap extends Transducer {
...
 }
+
+export const interpose = separator => xf => new Interpose(separator, xf)
+class Interpose extends Transducer {
+  constructor(separator, xf) {
+    super(xf)
+    this.separator = separator
+    this.started   = false
+  }
+  [tStep](result, value) {
```
Comment (kevinbeaty) [inline src/transducers.js]: I've been using `(value, input)` here.  The `input` taken from the "input source" as described in Clojure's documentation, and wanted something different than `result` to distinguish it from the `result` function.  Likewise `[tResult](value)`.  I'm open to changing it, but would like to be consistent.

**Comment by tgriesser** [comment]
Sure

**Comment by tgriesser** [comment]
Updated with docs & code style changes

**Pull request merged**